src/System.Management.Automation/engine/MshMemberInfo.cs

```diff
@@ -2642,8 +2643,326 @@ public override string TypeNameOfValue
+                case 12: return typeof(PSGenericType12);
+                case 13: return typeof(PSGenericType13);
+                case 14: return typeof(PSGenericType14);
+                case 15: return typeof(PSGenericType15);
+                case 16: return typeof(PSGenericType16);
+            }
+            throw Assert.Unreachable;
+        }
+    }
+
+    class PSGenericType0 : PSGenericType { private PSGenericType0() { } }
+    class PSGenericType1 : PSGenericType { private PSGenericType1() { } }
+    class PSGenericType2 : PSGenericType { private PSGenericType2() { } }
+    class PSGenericType3 : PSGenericType { private PSGenericType3() { } }
+    class PSGenericType4 : PSGenericType { private PSGenericType4() { } }
+    class PSGenericType5 : PSGenericType { private PSGenericType5() { } }
+    class PSGenericType6 : PSGenericType { private PSGenericType6() { } }
+    class PSGenericType7 : PSGenericType { private PSGenericType7() { } }
+    class PSGenericType8 : PSGenericType { private PSGenericType8() { } }
+    class PSGenericType9 : PSGenericType { private PSGenericType9() { } }
+    class PSGenericType10 : PSGenericType { private PSGenericType10() { } }
+    class PSGenericType11 : PSGenericType { private PSGenericType11() { } }
+    class PSGenericType12 : PSGenericType { private PSGenericType12() { } }
+    class PSGenericType13 : PSGenericType { private PSGenericType13() { } }
+    class PSGenericType14 : PSGenericType { private PSGenericType14() { } }
+    class PSGenericType15 : PSGenericType { private PSGenericType15() { } }
+    class PSGenericType16 : PSGenericType { private PSGenericType16() { } }
+
+    internal abstract class MethodGroup { }
+    internal class MethodGroup<T1> : MethodGroup { }
+    internal class MethodGroup<T1, T2> : MethodGroup { }
+    internal class MethodGroup<T1, T2, T3, T4> : MethodGroup { }
+    internal class MethodGroup<T1, T2, T3, T4, T5, T6, T7, T8> : MethodGroup { }
+    internal class MethodGroup<T1, T2, T3, T4, T5, T6, T7, T8, T9, T10, T11, T12, T13, T14, T15, T16> : MethodGroup { }
+    internal class MethodGroup<T1, T2, T3, T4, T5, T6, T7, T8, T9, T10, T11, T12, T13, T14, T15, T16, T17, T18, T19, T20, T21, T22, T23, T24, T25, T26, T27, T28, T29, T30, T31, T32> : MethodGroup { }
+
+    class Unit
+    {
+        private Unit() { }
+    }
+
+
+    internal struct PSMethodSignatureEnumerator : IEnumerator<Type>
+    {
+        private int _current;
+        private readonly Type _t;
+
+        internal PSMethodSignatureEnumerator(Type t)
+        {
+            Diagnostics.Assert(t.IsSubclassOf(typeof(PSMethod)), "Must be a PSMethod<MethodGroup<>>");
+            _t = t.GenericTypeArguments[0];
+            Current = null;
+            _current = -1;
+        }
+
+        public bool MoveNext()
+        {
+            _current++;
+            return MoveNext(_t, _current);
+        }
+
+        bool MoveNext(Type type, int index)
```

Comment:
Public? Private?

Reply:
Is there a good reason for this private method? It's only called in one place.